atom/browser/net/url_request_context_getter.cc

```diff
@@ -328,6 +357,31 @@ net::URLRequestContext* URLRequestContextGetter::GetURLRequestContext() {
         main_request_context_->transport_security_state();
     transport_security_state->SetRequireCTDelegate(ct_delegate_.get());
 
+    auto* cookie_monster =
+        static_cast<net::CookieMonster*>(main_request_context_->cookie_store());
+    std::vector<std::string> cookie_schemes(
+        {url::kHttpScheme, url::kHttpsScheme, url::kWsScheme, url::kWssScheme});
+    const auto& custom_standard_schemes = atom::api::GetStandardSchemes();
+    cookie_schemes.insert(cookie_schemes.end(), custom_standard_schemes.begin(),
+                          custom_standard_schemes.end());
+    cookie_monster->SetCookieableSchemes(cookie_schemes);
+
+    top_job_factory_ =
+        CreateURLRequestJobFactory(&protocol_handlers_, main_request_context_);
+    std::unique_ptr<net::URLRequestJobFactory> inner_job_factory(
+        new net::URLRequestJobFactoryImpl);
+    if (!protocol_interceptors_.empty()) {
+      // Set up interceptors in the reverse order.
+      for (auto it = protocol_interceptors_.rbegin();
+           it != protocol_interceptors_.rend(); ++it) {
+        inner_job_factory.reset(new net::URLRequestInterceptingJobFactory(
+            std::move(inner_job_factory), std::move(*it)));
+      }
+      protocol_interceptors_.clear();
+    }
+    top_job_factory_->Chain(std::move(inner_job_factory));
+    main_request_context_->set_job_factory(top_job_factory_.get());
```

Comment:
The ownership is transferred here. How will the lifetimes work for `top_job_factory_`?

Reply:
Its not transferred, `URLRequestContext` only has weak references to its members, any reason to think the ownership was transferred here ?